server/src/main/java/org/opensearch/common/blobstore/BlobContainer.java

```diff
@@ -124,6 +124,15 @@ default long readBlobPreferredLength() {
      */
     void writeBlob(String blobName, InputStream inputStream, long blobSize, boolean failIfAlreadyExists) throws IOException;
 
+    /**
+     * Needs to be set to true by the implementation if remote data integrity check is supported
+     *
+     * @return true if remote data integrity check is supported
+     */
+    default boolean isRemoteDataIntegritySupported() {
+        return false;
+    }
+
```

Comment:
Can't we check if the Stream is an instance `ResettableCheckedInputStream` and accordingly derive from it. Do we really need this?

Reply:
We construct `ResettableCheckdInputStream` on the basis of `isRemoteDataIntegritySupported` which has to be set to true by the vendor plugin manually. If we remove this, we have no way of knowing if the plugin supports server side data integrity check.

Reply:
Let's use something like VerifyingMultiStreamBlobContainer interface